hazelcast/proxy/vector_collection.py

```diff
@@ -0,0 +1,282 @@
+            future = self._invoke_on_partition(request, partition_id)
+            futures.append(future)
+
+        return combine_futures(futures)
+
+    def put_if_absent(self, key: Any, document: Document) -> Future[Optional[Document]]:
+        check_not_none(key, "key can't be None")
+        check_not_none(document, "document can't be None")
+        check_not_none(document.value, "document value can't be None")
+        return self._put_if_absent_internal(key, document)
+
+    def search_near_vector(
+        self,
+        vector: Vector,
+        *,
+        include_value: bool = False,
+        include_vectors: bool = False,
+        limit: Optional[int] = None
+    ) -> Future[List[SearchResult]]:
+        check_not_none(vector, "vector can't be None")
+        if limit is None:
+            limit = 2147483629
+        return self._search_near_vector_internal(
+            vector,
+            include_value=include_value,
+            include_vectors=include_vectors,
+            limit=limit,
+        )
+
+    def remove(self, key: Any) -> Future[Optional[Document]]:
+        check_not_none(key, "key can't be None")
+        return self._remove_internal(key)
+
+    def delete(self, key: Any) -> Future[None]:
+        check_not_none(key, "key can't be None")
+        return self._delete_internal(key)
+
+    def _set_internal(self, key: Any, document: Document) -> Future[None]:
+        try:
+            key_data = self._to_data(key)
+            value_data = self._to_data(document.value)
+        except SchemaNotReplicatedError as e:
+            return self._send_schema_and_retry(e, self.set, key, document)
+        document = copy.copy(document)
+        document.value = value_data
+        request = vector_collection_set_codec.encode_request(
+            self.name,
+            key_data,
+            document,
+        )
+        return self._invoke_on_key(request, key_data)
+
+    def _get_internal(self, key: Any):
+        def handler(message):
+            doc = vector_collection_get_codec.decode_response(message)
+            return self._transform_document(doc)
+
+        try:
+            key_data = self._to_data(key)
+        except SchemaNotReplicatedError as e:
+            return self._send_schema_and_retry(e, self.get, key)
+        request = vector_collection_get_codec.encode_request(
+            self.name,
+            key_data,
+        )
+        return self._invoke_on_key(request, key_data, response_handler=handler)
+
+    def _search_near_vector_internal(
+        self,
+        vector: Vector,
+        *,
+        include_value: bool = False,
+        include_vectors: bool = False,
+        limit: Optional[int] = None
+    ) -> Future[List[SearchResult]]:
+        def handler(message):
+            results: List[
+                SearchResult
+            ] = vector_collection_search_near_vector_codec.decode_response(message)
+            for result in results:
```

Comment:
what about setting vector.type here? SearchResult includes vector list

Reply:
Yes, that's a mistake.
Fixed at: https://github.com/hazelcast/hazelcast-python-client/pull/674/commits/560871086ac734fcd0a044da584a8309604daa98